CMakeLists.txt

```diff
@@ -0,0 +1,250 @@
+# SPDX-License-Identifier: Unlicense
+#
+# LibTomMath, a free open source portable number theoretic multiple-precision
+# integer (MPI) library written entirely in C.
+#
+
+cmake_minimum_required(VERSION 3.10)
+
+project(tommath VERSION 1.2.0)
+
+# package release version
+# bump if re-releasing the same VERSION + patches
+# set to 1 if releasing a new version
+set(PACKAGE_RELEASE_VERSION 1)
+
+#-----------------------------------------------------------------------------
+# Include cmake modules
+#-----------------------------------------------------------------------------
+include(GNUInstallDirs)
+include(CMakePackageConfigHelpers)
+include(sources.cmake)
+
+# The only direct cmake argument for now
+option(BUILD_SHARED_LIBS "Build shared library and only the shared library if \"ON\", default is static" OFF)
+
+#-----------------------------------------------------------------------------
+# Compose CFLAGS
+#-----------------------------------------------------------------------------
+
+# Some information copied from makefile_include.mk
+
+# Basic set
+set(LTM_C_FLAGS  -Wall -Wsign-compare -Wextra -Wshadow)
+set(LTM_C_FLAGS  ${LTM_C_FLAGS} -Wdeclaration-after-statement -Wbad-function-cast -Wcast-align)
+set(LTM_C_FLAGS  ${LTM_C_FLAGS} -Wstrict-prototypes -Wpointer-arith -Wsystem-headers)
+
+if(COMPILE_DEBUG)
```

Comment:
Probably [CMAKE_BUILD_TYPE](https://cmake.org/cmake/help/v3.10/variable/CMAKE_BUILD_TYPE.html#cmake-build-type) should be used for this kind of thing.

CMAKE_C_FLAGS_[DEBUG|RELEASE|RELWITHDEBINFO|MINSIZEREL] variables are used depending on the build type with the following default values

```
CMAKE_C_FLAGS_DEBUG: -g
CMAKE_C_FLAGS_RELEASE: -O3 -DNDEBUG
CMAKE_C_FLAGS_MINSIZEREL: -Os -DNDEBUG
CMAKE_C_FLAGS_RELWITHDEBINFO: -O2 -g -DNDEBUG
```

Default values can be overriden e.g. for -g3 in debug builds `set(CMAKE_C_FLAGS_DEBUG -g3)`

Reply:
Fine like that?